### DISCUSSION TITLE
PrefixCacheTask is overwriting prefixes on redis sessions

### DISCUSSION THREAD

**Discussion by recognizerHD** [spatie/laravel-multitenancy#311] [Q&A]
Hello, I've been migrating my project to use multitenancy and I think I'm stuck on one final piece. I can't use the EnsureValidTenantSession as the session should be tenant unaware. A single user could manage several tenants and switch easily between them. I've got this almost working except when it saves the session, it prefixes the session key with the tenant information: 

![image](https://user-images.githubusercontent.com/1222899/144162450-c04d2d22-9687-4f8d-8d66-9e066c33de90.png)

I'm using redis for sessions and it looks like the PrefixCacheTask is what replaces the prefix. I do wish to keep the class as other things within the tenant databases will be cached and I would like to keep this functionality.

If you could point me in the right direction, that would be great.

**Comment by masterix21** [comment]
I think you should create a custom middleware to check the session, for example, creating a session with all user's tenant ids.

**Comment by recognizerHD** [comment]
So through some trial and error, I think I got it working.

In config/database.php I added another entry for redis and copied the same info as the cache, but changed the DB it was pointed to.
```
        'session' => [
            ...
            'database' => env('REDIS_SESSION_DB', '0'),
//            'prefix'   => ..., // no prefix. defined in cache
        ],
```

In config/cache.php I added a new store:
```
        'redis_session' => [
            'driver'     => 'redis',
            'connection' => 'session',
            'prefix' => Str::slug(env('APP_NAME', 'laravel'), '_').'_session'
        ],
```

 It used the connection that I created in database and it was also a redis driver. Gave it its own prefix value here.

Then finally in .env I added 
```
SESSION_CONNECTION=session
SESSION_STORE=redis_session
```

Without any of these, it didn't work. Hope this helps anyone with the same issue as me.

BTW. Huge thank you for making the project in the first place. I had initially started out making my own but there was so much missing functionality. And an issues board with plenty of issues but they are all resolved/closed is a thing of beauty.
